### DISCUSSION TITLE
"Dependencies do not meet ready condition" notification

### DISCUSSION THREAD

**Discussion by nicogl** [fluxcd/flux2#1160] [Q&A] [answered]
I set up a couple of Kustomizations, one with some infrastructure-base components (controllers) and the other one for the main application.

Everything is working fine; the application is running and the continuous delivery pipeline implemented by Flux works as expected when changes are done/commited.

However, I'm getting some weird notifications in my Slack channel every now and then... for example I got below notification once every three hours approximately:

![image](https://user-images.githubusercontent.com/10395658/112571348-9e12b800-8dc6-11eb-8bcb-1dfa1d16c77a.png)

Note the green color of the vertical bar in the Slack notification... as per my understanding this means that it's not an error.

And if I check the "kustomize-controller" pod's log, I can see messages like this:

`{"level":"error","ts":"2021-03-26T02:39:43.980Z","logger":"controller.kustomization","msg":"Dependencies do not meet ready condition, retrying in 30s","reconciler group":"kustomize.toolkit.fluxcd.io","reconciler kind":"Kustomization","name":"apps","namespace":"flux-system","error":"dependency 'flux-system/infrastructure' is not ready"}`

Note that I didn't make any change in between notifications or since I deployed the application for the first time. Also, as I said before, the application runs fine and seems that there's no outage.

Any idea what may be causing this notification to trigger?

**Comment by allenporter** [comment]
I see something similar as well. I'm happy to provide additional information, though not sure exactly where to dig deeper.

**Comment by nicogl** [comment, marked as answer]
Hi @stefanprodan and @hiddeco 
Any suggestion that you can provide on this? Thanks in advance

**Reply by stefanprodan** [reply]
The log level is wrong, should be info as this a just a to let you know that the dependency is reconciling so no action can be taken.

**Reply by allenporter** [reply]
I see in the code the event is sent with Info but the error is logged with Error. I   I can send out a PR to update this from Error to Info, looking for an excuse to learn about flux2, if nobody has already done it.
